Libs/MRML/Core/vtkMRMLPlotLayoutNode.h

```diff
@@ -0,0 +1,261 @@
+
+  ///
+  /// Set node attributes
+  virtual void ReadXMLAttributes( const char** atts);
+
+  ///
+  /// Write this node's information to a MRML file in XML format.
+  virtual void WriteXML(ostream& of, int indent);
+
+  ///
+  /// Copy the node's attributes to this object
+  virtual void Copy(vtkMRMLNode *node);
+
+  ///
+  /// Get node XML tag name (like Volume, Model)
+  virtual const char* GetNodeTagName() {return "PlotLayout";};
+
+  ///
+  /// Method to propagate events generated in mrml
+  virtual void ProcessMRMLEvents ( vtkObject *caller, unsigned long event, void *callData );
+
+  //----------------------------------------------------------------
+  /// Access methods
+  //----------------------------------------------------------------
+
+  ///
+  /// Add and Observe a Plot to the PlotLayout. Parameter "name" is used for
+  /// referencing the Plot when setting properties for plotting the
+  /// Plot or for removing the Plot from the PlotLayout.
+  virtual void AddAndObservePlot(const char *name, const char * id);
+
+  ///
+  /// Remove a Plot and Observation from the PlotLayout by a particular name
+  virtual void RemovePlotAndObservationByName(const char *name);
+
+  ///
+  /// Remove a Plot and Observation from the PlotLayout by ID
+  virtual void RemovePlotAndObservationByID(const char *id);
+
+  ///
+  /// Update all missing Observations
+  virtual void UpdateObservations();
+
+  ///
+  /// Update a Plot Observation if is missing
+  virtual void UpdateObservation(const char * id);
+
+  ///
+  /// Remove all the Plots
+  virtual void ClearPlotIDs();
+
+  ///
+  /// Get the Plot id referenced by a particular name
+  virtual const char* GetPlotID(const char *name);
+
+  ///
+  /// Get the Plot referenced by a particular name
+  virtual vtkMRMLPlotNode* GetPlotbyName(const char *name);
+
+  ///
+  /// Get the Plot referenced by a particular id
+  virtual vtkMRMLPlotNode* GetPlotByID(const char *id);
+
+  ///
+  /// Get the list of Plot names
+  virtual vtkStringArray* GetPlotNames();
+  virtual int GetPlotNames(std::vector<std::string> &plotNodeNames);
+
+  ///
+  /// Get the list of Plot ids
+  virtual vtkStringArray* GetPlotIDs();
+  virtual int GetPlotIDs(std::vector<std::string> &plotNodeIDs);
+
+  ///
+  /// Get the collection of PlotNodes
+  virtual vtkCollection* GetPlotNodes();
+
+  ///
+  /// Set/Get the selected PlotNodeIDs
+  virtual void SetSelectionPlotNodeIDs(vtkStringArray* selectedPlotNodeIDs);
+  virtual vtkStringArray* GetSelectionPlotNodeIDs();
+
+  ///
+  /// Set/Get the collection of selectedIDArrays
+  virtual void SetSelectionArrays(vtkCollection* selectedArrays);
+  virtual vtkCollection *GetSelectionArrays();
+
+  ///
+  /// Set the PlotNodeIDs and selectedArrays
+  virtual void SetSelection(vtkStringArray* selectedPlotNodeIDs,
+                            vtkCollection* selectedArrays);
+
+  ///
+  /// Get the index of a PlotNode
+  virtual vtkIdType GetPlotNodeIndex(vtkMRMLPlotNode* plotNode);
```

Comment:
is the "index" of teh plot in plotNodes collection. But plotNodes should go away as well updating to the reference infrastructure